block/manager.go

```diff
@@ -181,21 +175,17 @@ func NewManager(
 		HeaderCh:          make(chan *types.SignedHeader, channelLength),
 		BlockCh:           make(chan *types.Block, channelLength),
 		blockInCh:         make(chan newBlockEvent, blockInChLength),
-		blockStoreMtx:     new(sync.Mutex),
+		blockStoreCh:      make(chan struct{}),
 		blockStore:        blockStore,
-		retrieveMtx:       new(sync.Mutex),
 		lastStateMtx:      new(sync.RWMutex),
 		blockCache:        NewBlockCache(),
+		retrieveCh:        make(chan struct{}),
```

Comment:
This channel is unbuffered - which means that `SyncLoop` can be blocked by `RetrieveLoop`. Probably this won't be visible in unit tests, but as soon as DA layer slows down or doesn't respond problem can occur.

Scenario:
1. Message is being send to `retrieveCh` in `SyncLoop`.
2. `RetrieveLoop` works slowly, because it needs retries, DA responds slowly, etc.
3. `SyncLoop` sends a message to `retrieveCh` before previous step is finished -> now `SyncLoop` is blocking on `RetrieveLoop`, and this is not desirable.

Reply:
Great catch! I think making the channel buffered only delays the issue as well. Do you think the best thing here is to just revert the code to the over-engineered version here?

Reply:
Guess this can be tested with mock DA by simulating slow reads.

Did a quick test against `main` and seems like even aggregation loop is being blocked, when DA is slow / non-responsive. Perhaps would be good to add tests for this as connection / node sync would be quite common in real world usage.

FWIW aggregation with slow DA seems to be working with this PR.

Reply:
Seems like aggregation with slow DA was fixed by `pendingBlocks` refactor.

Reply:
We don't need to revert to the overengineered version imo. We just need to handle this case and not make it blocking. 

We can add a buffer of 1 to the chan and then make the calling loops non-blocking on that signalling.

Example: https://go.dev/play/p/L_GwbNfSrB2

The one caveat is that this design assuming the calling loops will continue to call and supply pressure to the system, so that the receiving loops will always have something to pull from in the channel. 

I think this is a safe assumption for this case, because if this wasn't the case, then we'd need to figure out why we aren't producing anything.

Reply:
Thanks @tuxcanfly and @MSevey, great feedback! Thought about this more and this makes a lot more sense:

> We can add a buffer of 1 to the chan and then make the calling loops non-blocking on that signaling.

Implemented this change, feel free to review it again and resolve it if it seems sufficient.